### PR TITLE
fix version number for tarball packages

### PR DESCRIPTION
**Pull request by shahasad**

### CODE REVIEW

**Comment by pranavsharma** [comment]
Something doesn't seem right. The artifact name produced is "onnxruntime-win-x64-0.3.0.$(OnnxRuntimeGitCommitHash).zip". Two things: 
1. The git commit hash didn't get expanded.
2. I don't think we should put the git commit hash in the name of the pkg file. This will make the package name too long. Even our nuget pkg doesn't do this today. The nuget pkg file name is microsoft.ml.onnxruntime.0.2.1.nupkg and the commit hash is present inside this pkg. We can write the commit hash to a file and include it as part of the pkg. For example tensorflow names it's zip file as libtensorflow-cpu-windows-x86_64-1.12.0.zip (Notice the version number in the filename, without the commit hash - alas tensorflow doesn't include the commit hash in the pkg).

File: tools/ci_build/github/azure-pipelines/templates/set-version-number-variables-step.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+steps:
+- task: CmdLine@2
+  displayName: 'Set Version Number variables for  Windows'
+  inputs:
+    script: |
+        SETLOCAL EnableDelayedExpansion
+        set /p _OnnxRuntimeVersion=<$(Build.SourcesDirectory)\VERSION_NUMBER 
+        @echo ##vso[task.setvariable variable=OnnxRuntimeVersion;]%_OnnxRuntimeVersion%
+        
+        FOR /F "tokens=* USEBACKQ" %%F IN (`git rev-parse --short HEAD`) DO (
```
Comment (jignparm) [inline tools/ci_build/github/azure-pipelines/templates/set-version-number-variables-step.yml]: This gives the short version of the commit hash. Do we want to retain the full hash instead? Or is short version sufficient?

**Comment by shahasad** [comment]
> Something doesn't seem right. The artifact name produced is "onnxruntime-win-x64-0.3.0.$(OnnxRuntimeGitCommitHash).zip". Two things:
> 
> The git commit hash didn't get expanded.

This is fixed now.


> I don't think we should put the git commit hash in the name of the pkg file. This will make the package name too long. Even our nuget pkg doesn't do this today. The nuget pkg file name is microsoft.ml.onnxruntime.0.2.1.nupkg and the commit hash is present inside this pkg. We can write the commit hash to a file and include it as part of the pkg. For example tensorflow names it's zip file as libtensorflow-cpu-windows-x86_64-1.12.0.zip (Notice the version number in the filename, without the commit hash - alas tensorflow doesn't include the commit hash in the pkg).

OK, I will make change accordingly then.

File: tools/ci_build/github/azure-pipelines/templates/set-version-number-variables-step.yml (hunk 1 above)
Comment (shahasad) [reply tools/ci_build/github/azure-pipelines/templates/set-version-number-variables-step.yml]: if kept in an internal file, will use the long commit hash

File: tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml [hunk 2]
```diff
@@ -3,52 +3,61 @@ jobs:
...
       parameters:
           buildConfig: 'Release'
-          artifactName: 'onnxruntime-linux-x64'
+          artifactName: 'onnxruntime-linux-x64-$(OnnxRuntimeVersion)'
           libraryName: 'libonnxruntime.so'
+          commitId: $(OnnxRuntimeGitCommitHash)
     - template: templates/clean-agent-build-directory-step.yml
 
 
 - job: Linux_C_API_Packaging_CPU_x86
```
Comment (RandySheriffH) [inline tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml]: Please merge the c-api yml file with latest release.

**Pull request merged**